devtools/maven/src/main/java/io/quarkus/maven/UpdateToLatestVersionMojo.java

```diff
@@ -0,0 +1,132 @@
+package io.quarkus.maven;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Updates the Quarkus version properties in the pom.xml file to the latest version.
+ * 
+ * It is retrieved from the release page on Github.
+ */
+@Mojo(name = "update-to-latest")
+public class UpdateToLatestVersionMojo extends AbstractMojo {
+
+    private static final String QUARKUS_PLATFORM_VERSION = "quarkus.platform.version";
+    private static final String QUARKUS_PLUGIN_VERSION = "quarkus-plugin.version";
+
+    @Parameter(property = QUARKUS_PLATFORM_VERSION, required = true)
+    private String platformVersion;
+
+    @Parameter(property = QUARKUS_PLUGIN_VERSION, required = true)
+    private String pluginVersion;
+
+    @Parameter(property = "community", defaultValue = "true")
+    private boolean isCommunity;
+
+    private String pomfile = "pom.xml";
+    private File pomFile = new File(pomfile);
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        if (!isCommunity) {
+            throw new MojoExecutionException("not implemented yet");
+        }
+
+        MavenXpp3Reader mavenreader = new MavenXpp3Reader();
+        Model model;
+        try {
+            FileReader reader = new FileReader(pomFile);
+            model = mavenreader.read(reader);
+            model.setPomFile(pomFile);
+        } catch (Exception e) {
+            throw new MojoExecutionException(e.getMessage());
+        }
+        MavenProject project = new MavenProject(model);
```

Comment:
You  can inject the MavenProject directly in your mojo

Reply:
I know, the idea was to have the existing POM file on disk without anything being resolved, transitively added and so on. I'm not 100% sure if this is the case if I inject it. I'll have a detailed look at that.